File: test-plugins/org.yakindu.sct.generator.java.runtime.test/src-gen/org/yakindu/sct/generator/java/runtime/test/GuardTest.java
/**
 * Copyright (c) 2012 committers of YAKINDU and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *     committers of YAKINDU - initial API and implementation
 */
package org.yakindu.sct.generator.java.runtime.test;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;
import org.yakindu.scr.guard.GuardStatemachine;
import org.yakindu.scr.guard.GuardStatemachine.State;
/**
 *  Unit TestCase for Guard
 */
@SuppressWarnings("all")
public class GuardTest {

	private GuardStatemachine statemachine;

	@Before
	public void setUp() {
		statemachine = new GuardStatemachine();
		statemachine.init();
		statemachine.enter();
	}

	@After
	public void tearDown() {
		statemachine = null;
	}

	@Test
	public void testguardTest() {
		assertTrue(statemachine.isStateActive(State.Main_region_A));
	}
}

File: test-plugins/org.yakindu.sct.generator.c.test/gtests/GuardTest/GuardTest.cc
/**
* Copyright (c) 2017 committers of YAKINDU and others.
* All rights reserved. This program and the accompanying materials
* are made available under the terms of the Eclipse Public License v1.0
* which accompanies this distribution, and is available at
* http://www.eclipse.org/legal/epl-v10.html
*
* Contributors:
*     committers of YAKINDU - initial API and implementation
*/
#include "gtest/gtest.h"
#include "Guard.h"

Guard handle;

class StatemachineTest : public ::testing::Test{
	protected:
	virtual void SetUp() {
		guard_init(&handle);
	}
};


TEST_F(StatemachineTest, guardTest) {					
	guard_enter(&handle);
	EXPECT_TRUE(guard_isStateActive(&handle, Guard_main_region_A));
	guardIface_raise_event1(&handle);
	guard_runCycle(&handle);
	EXPECT_TRUE(guard_isStateActive(&handle, Guard_main_region_A));
	guardIface_raise_event2(&handle);
	guard_runCycle(&handle);
	EXPECT_TRUE(guard_isStateActive(&handle, Guard_main_region_B));
	guardIface_raise_return(&handle);
	guard_runCycle(&handle);
	EXPECT_TRUE(guard_isStateActive(&handle, Guard_main_region_A));
	guardIface_raise_event1(&handle);
	guard_runCycle(&handle);
	EXPECT_TRUE(guard_isStateActive(&handle, Guard_main_region_B));
	guardIface_raise_return(&handle);
	guard_runCycle(&handle);
	EXPECT_TRUE(guard_isStateActive(&handle, Guard_main_region_A));
}


